edb/server/protocol/binary.pyx

```diff
@@ -667,22 +686,18 @@ cdef class EdgeConnection(frontend.FrontendConnection):
             ) from None
 
         if not claims.get(f"{namespace}.any_role"):
+            assert not is_legacy  # legacy CLI will only generate any_role JWTs
             token_roles = claims.get(f"{namespace}.roles")
-            if not isinstance(token_roles, dict):
+            if not isinstance(token_roles, list):
```

Comment:
Legacy tokens would still have a dict here, no?

Reply:
Yes, that's why I put an assertion above - I think we could also do:

```suggestion
            if not isinstance(token_roles, dict if is_legacy else list):
```

Reply:
Well, yes, otherwise it will crash on a legacy token.